src/api.ts

```diff
@@ -29,7 +29,7 @@ export interface MetadataSubmission {
   templates: string[];
   studyForm: StudyForm;
   multiOmicsForm: object;
-  sampleData: object;
+  sampleData: Record<string, Record<string, string>[]>;
```

Comment:
I don't know whether you got this TypeScript type definition from `nmdc-server`. I recently came to the conclusion that at least one of the type definitions currently in use in `nmdc-server` for a JSON value is inadequate. I filed an issue about that [here](https://github.com/microbiomedata/nmdc-server/issues/1142). 

I proposed this generic JSON value type instead:

```ts
/**
 * Define a TypeScript type alias that accounts for all possible JSON values.
 *
 * References:
 * - https://www.json.org/json-en.html
 * - https://devblogs.microsoft.com/typescript/announcing-typescript-3-7/#more-recursive-type-aliases
 */
export type JSONValue = string | number | boolean | null | { [k: string]: JSONValue } | Array<JSONValue>;
```

Reply:
Technically it should adhere to the `SampleData` class defined in the submission-schema: https://github.com/microbiomedata/submission-schema/blob/fff9173fc4aa42a609fcb7e86b3f71c0a10d5128/src/nmdc_submission_schema/schema/nmdc_submission_schema.yaml#L51005-L51024. So it's a little more restricted than just any JSON value. This is sort of a half-step towards that. Once we get into sample editing we'll need to integrate the submission schema. At that point we could also look into [generating TypeScript](https://linkml.io/linkml/generators/typescript.html) from the LinkML schema.

Reply:
OK, thanks for clarifying. I'm OK with this being merged in as is, given that we will be revisiting it at a later stage of development.